src/de/schildbach/pte/AvvAachenProvider.java

```diff
@@ -48,6 +48,12 @@ public AvvAachenProvider(final String apiClient, final String apiAuthorization)
 
     @Override
     protected String[] splitStationName(final String name) {
+        // Some stations in Aachen city have both the names "station name, AC" and "AC, station name".
+        // If you type the station name first, you get the variant "station name, AC", which would be parsed as the
+        // station AC in "station name".
+        if (name.endsWith(", AC")) {
+            return new String[] { "Aachen" , name.substring(0, name.length() - ", AC".length()) };
+        }
```

Comment:
I'm not sure, but shouldn't both of the above variants be split into "Aachen" and the plain station name?

Reply:
I don't care (They are already split into "AC" and plain station name).

Reply:
You added another filter for that